### PR TITLE
CMEK: now with 200% more Vault!

### PR DESCRIPTION
**Pull request by MichaelTrestman**
https://cockroachlabs.atlassian.net/browse/DOC-3387

### CODE REVIEW

File: cockroachcloud/cmek-ops-gcp.md [hunk 1]
```diff
@@ -9,6 +9,18 @@ This page covers the procedures required to provision a Customer Managed Encrypt
 
 This is part of the larger process of [Enabling CMEK for a {{ site.data.products.dedicated }} cluster](managing-cmek.html#enable-cmek).
 
+## Overview
+
+- In [Step 1. Provision the cross-tenant service account](#step-1-provision-the-cross-tenant-service-account), we will create a service account that will be used by {{ site.data.products.dedicated }} to access the CMEK key.
```
Comment (kathancox) [inline cockroachcloud/cmek-ops-gcp.md:14]: Perhaps this sentence could be adjusted to "... we will create a service account that {{ site.data.products.dedicated }} will use to access the CMEK key." To remove the passive.

File: cockroachcloud/cmek-ops-gcp.md [hunk 2]
```diff
@@ -71,11 +86,106 @@ Here we will create a cross-tenant service account that can be temporarily assum
 	1. **Protection level**: **Software** 
 	1. **Purpose**: **Symmetric encrypt/decrypt**
 
+After you have provisioned the IAM role and KMS key for your CockroachDB cluster's CMEK, return to [Enabling CMEK for a {{ site.data.products.dedicated }} cluster](managing-cmek.html#step-4-enable-cmek-for-your-cockroachdb-dedicated-cluster).
+
+### Option B: Use the Vault GCP-KMS secrets engine to create the CMEK key
+
+Pre-requisites:
```
Comment (kathancox) [inline cockroachcloud/cmek-ops-gcp.md:93]: "Pre-requisites" --> "Prerequisites"

**Review by kathancox — APPROVED** [review]
LGTM with some minor nits / typos

**Review by adwittumuluri — APPROVED** [review]
I've read through this and the steps make sense, but I have no hands-on experience with the vault KMS secrets engine. By all accounts, vault-managed KMS-es should work identically to vanilla cloud provider ones, but I'd recommend (either you or me) testing.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14287#-:-N4dGLZ3CH0BZypF-4Zq:b-x9dr4a)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox and @MichaelTrestman)

**Review by MichaelTrestman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14287)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox and @MichaelTrestman)

___
*[`cockroachcloud/cmek-ops-aws.md` line 17 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7s-r2-17:-N4gqlC7EDBPH4foDkxs:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-aws.md#L17)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Typo: "with to AWS KMS" I think you intended to add "access" in between per the lower repeat of this sentence?
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-aws.md` line 87 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7s-r2-87:-N4gxcEL1a5uHQJmsYXT:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-aws.md#L87)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Should we link out for the two prerequisites here?
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-aws.md` line 92 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7s-r2-92:-N4gyZueAPmQIrYq9JkF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-aws.md#L92)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Switch out the backticks here for ~ on the code blocks per the usual markdown style we use (and the lower code block)
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-aws.md` line 153 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7s-r2-153:-N4gyakj0eMGSowi1MCf:b-i29r4d) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-aws.md#L153)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Out of curiosity: why did you decide to have an appendix rather than just have this listed in (9)? Will you potentially link to this from elsewhere in the docs?
</blockquote></details>

There are two links to it, from option A and option B, that's all. It was either that or write it out twice or link from one option to the other, appendix seemed the best of the 3. open to other suggestions.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 14 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7t-r2-14:-N4gzBJF8ck5ETmYcFSy:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-gcp.md#L14)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Perhaps this sentence could be adjusted to "... we will create a service account that {{ site.data.products.dedicated }} will use to access the CMEK key." To remove the passive.
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 93 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7t-r2-93:-N4gzSu85fZDyclRt8PL:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-gcp.md#L93)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

"Pre-requisites" --> "Prerequisites"
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 100 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7t-r2-100:-N4gzWLyCSSk5jKaPopv:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-gcp.md#L100)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Add a period to the end of this bullet point.
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 124 at r2](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4bkg9U4ilFc4tZsx7t-r2-124:-N4gzX3QAJdo4ZdLdBi0:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b94c998bbb2db7e471fa6d576ef50aeaef221d5d/cockroachcloud/cmek-ops-gcp.md#L124)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Can you edit the backticks in the code blocks on this page too, to match our general style using ~~~?
</blockquote></details>

Done.

**Comment by adwittumuluri** [comment]
verified vault+GCP KMS works in staging! :lgtm_strong:

**Comment by adwittumuluri** [comment]
sorry, in production, not staging* :lgtm_strong:

**Review by mdlinville — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14287)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kathancox, @mdlinville, and @MichaelTrestman)

___
*[`cockroachcloud/cmek-ops-aws.md` line 20 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4iyMsv45Ihz94FGVD6:-N4iyMsv45Ihz94FGVD7:belk06k) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L20)):*
> ```Markdown
> 
> {{site.data.alerts.callout_info}}
> For multi-region clusters, you must provide a key and IAM role combination per region. You can provide the same key for all your cluster regions, a different key per cluster region, or any mapping of keys to regions you may choose. It does not matter if the key is a single- or multi-region key.
> ```

Is it better to go with "cross-account IAM role" in all instances? Also, probably makes sense to go with just "region" after the first "cluster region"
___
*[`cockroachcloud/cmek-ops-aws.md` line 37 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4iyjcQ8DalkrMjxz-q:-N4iyjcQ8DalkrMjxz-r:b-m3r0g9) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L37)):*
> ```Markdown
> 1. Find your {{ site.data.products.dedicated }} cluster's associated  AWS Account ID.
> 
> 	You must find the Account ID of the AWS account that {{ site.data.products.dedicated }} will use for this purpose. To find the ID of the AWS account associated with your cluster, query the clusters endpoint of the {{ site.data.products.db }} API. The value is under the `account_id` field:
> ```

The `/clusters/` endpoint?
___
*[`cockroachcloud/cmek-ops-aws.md` line 76 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4iz4WH9CnzJs0ywyyK:-N4iz4WH9CnzJs0ywyyL:b-xc2kut) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L76)):*
> ```Markdown
> 1. For **Key usage**, specify **Encrypt and decrypt**.
> 1. Under **Advanced options**, choose **KMS** for **Key material**.
> 1. Select whether you will use your key for a single region or a multi-region cluster.
> ```

I think this is about whether the key is a single-region or multi-region key, right?
___
*[`cockroachcloud/cmek-ops-aws.md` line 77 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4izANZEB9g-AZ9LHnA:-N4izANZEB9g-AZ9LHnB:b-oa21fc) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L77)):*
> ```Markdown
> 1. Under **Advanced options**, choose **KMS** for **Key material**.
> 1. Select whether you will use your key for a single region or a multi-region cluster.
> 1. Give the key a suitable name, or **alias**.
> ```

Maybe clarify that this influences the key's URI and can't be changed, just in case they are less familiar with how that works?
___
*[`cockroachcloud/cmek-ops-aws.md` line 81 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4izMxjAz3RdB1VFUwL:-N4izMxjAz3RdB1VFUwM:b6h49tn) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L81)):*
> ```Markdown
> 1. Finish creating the key.
> 
> After you have provisioned the IAM role and KMS key for your CockroachDB cluster's CMEK, return to [Enabling CMEK for a {{ site.data.products.dedicated }} cluster](managing-cmek.html#step-4-enable-cmek-for-your-cockroachdb-dedicated-cluster).
> ```

Maybe "the cross-account IAM role and the CMEK key in AWS, return to...."
___
*[`cockroachcloud/cmek-ops-gcp.md` line 29 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j-8AZBunnvTqwWoRe:-N4j-8AZBunnvTqwWoRf:b8efd8f) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L29)):*
> ```Markdown
> - provide the same key for all your cluster regions.
> - provide a different key per cluster region.
> - provide a single dual- or multi-region key or global key for the entire cluster. See [GCP's Cloud KMS location docs](https://cloud.google.com/kms/docs/locations).
> ```

Interesting, in my PR I only got feedback that multi-region keys were supported on AWS. If they are also supported on GCP I need to add that info to the concept too.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 38 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j-Mxb5uFgBDs5clLw:-N4j-Mxb5uFgBDs5clLx:bkeyjey) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L38)):*
> ```Markdown
> 1. Find your {{ site.data.products.dedicated }} organization ID in the {{ site.data.products.db }} [organization settings page](https://cockroachlabs.cloud/settings).
> 
> 1. Find your {{ site.data.products.dedicated }} cluster ID:
> ```

Is it worth reusing this chunk in both the AWS and GCP topics? We might have other times when the reader needs to find the cluster ID.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 64 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j-by_8gm46Sq0iLBT:-N4j-by_8gm46Sq0iLBU:b-k7ppxu) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L64)):*
> ```Markdown
> 	
> 1. Authorize {{ site.data.products.dedicated }}'s GCP project to use the service account (making it *cross-tenant*).
> 	1. Click your newly created service account's email address to visit its details page.
> ```

nit: newly-created
___
*[`cockroachcloud/cmek-ops-gcp.md` line 168 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j0Hsg6zwBZjgTeO3R:-N4j0Hsg6zwBZjgTeO3S:b216w38) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L168)):*
> ```Markdown
> 	```
> 
> 1. Propagate the key to your KMS service
> ```

nit: missing a period

**Review by mdlinville — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14287)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kathancox, @mdlinville, and @MichaelTrestman)

___
*[`cockroachcloud/cmek-ops-gcp.md` line 29 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j-8AZBunnvTqwWoRe:-N5HQq0PBFXldjBRVxgP:b-2rrddc) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L29)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

Interesting, in my PR I only got feedback that multi-region keys were supported on AWS. If they are also supported on GCP I need to add that info to the concept too.
</blockquote></details>

I verified this does work on GCP (global or multi-regional), transparently.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 65 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N5HR2JX51QaNIswrvx2:-N5HR2JX51QaNIswrvx3:b-9l53yf) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L65)):*
> ```Markdown
> 1. Authorize {{ site.data.products.dedicated }}'s GCP project to use the service account (making it *cross-tenant*).
> 	1. Click your newly created service account's email address to visit its details page.
> 	1. Select the **PERMISSIONS** tab and click **ADD**.
> ```

I sent this feedback on Slack but don't want it to get lost -- the button isn't Add in this UI.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 70 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N5HRC0n8Q9i7iI68Y_k:-N5HRC0n8Q9i7iI68Y_l:b-rgkzfu) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L70)):*
> ```Markdown
> 		`crl-kms-user-{CLUSTER_ID}@{PROJECT_ID}.iam.gserviceaccount.com`
> 
> 		Where `CLUSTER_ID` is the **last 12 digits** of your cluster ID, and `PROJECT_ID` is the GCP project ID, which you obtained previously.
> ```

This tripped me up even having read it! It would be nice if we could just trim it for them if they used the whole cluster ID. No action required, I suppose.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 84 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N5HRqC88A_CY_rs-1kw:-N5HRqC9EZjQ2lH_iYDF:b-uhvbja) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L84)):*
> ```Markdown
> 1. In the GCP console, visit the [KMS page](https://console.cloud.google.com/security/kms).
> 1. Click **+ CREATE KEY RING** and fill in the details to complete the key ring.
> 1. Fill in the details and **CREATE** your encryption key:
> ```

Is it worth mentioning here that multi-regional or global keys are fine? Below, is it worth stating that you could just as easily be importing as generating the key?

**Review by MichaelTrestman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14287)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kathancox and @mdlinville)

___
*[`cockroachcloud/cmek-ops-aws.md` line 20 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4iyMsv45Ihz94FGVD6:-N5_bb9k4bnjYobKnpHH:b-7rl0jd) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L20)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

Is it better to go with "cross-account IAM role" in all instances? Also, probably makes sense to go with just "region" after the first "cluster region"
</blockquote></details>

done.
___
*[`cockroachcloud/cmek-ops-aws.md` line 37 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4iyjcQ8DalkrMjxz-q:-N5_YuQM-cxmCKgx4w45:bvpc7am) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L37)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

The `/clusters/` endpoint?
</blockquote></details>

Seems fine to me refer to it more loosely by function as the clusters endpoint?
___
*[`cockroachcloud/cmek-ops-aws.md` line 76 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4iz4WH9CnzJs0ywyyK:-N5_ZeXb-l4WdWgZcK-w:bfbq99i) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L76)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

I think this is about whether the key is a single-region or multi-region key, right?
</blockquote></details>

Good point, done.
___
*[`cockroachcloud/cmek-ops-aws.md` line 77 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4izANZEB9g-AZ9LHnA:-N5_ZOgA08T_fi3EQ9GM:b-7rl0jd) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L77)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

Maybe clarify that this influences the key's URI and can't be changed, just in case they are less familiar with how that works?
</blockquote></details>

done.
___
*[`cockroachcloud/cmek-ops-aws.md` line 81 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4izMxjAz3RdB1VFUwL:-N5_bwICBR5CUkGYFiAS:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-aws.md#L81)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

Maybe "the cross-account IAM role and the CMEK key in AWS, return to...."
</blockquote></details>

Done.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 29 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j-8AZBunnvTqwWoRe:-N5__7eH1hNTolDelUo_:btojbfd) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L29)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

I verified this does work on GCP (global or multi-regional), transparently.
</blockquote></details>

Great!
___
*[`cockroachcloud/cmek-ops-gcp.md` line 38 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N4j-Mxb5uFgBDs5clLw:-N5__Nzo4SPNSe1IR2wC:b2vigqd) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L38)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

Is it worth reusing this chunk in both the AWS and GCP topics? We might have other times when the reader needs to find the cluster ID.
</blockquote></details>

"worth it" is the operative question. for a chunk of a couple of sentences like this, i don't think so. If you extracted all of the re-used chunks this size in our docs into partials, debugging builds would be a bloody labyrinthian nightmare, and I think it slows down Jekyll's build process too, although I'm not 100% on that.
___
*[`cockroachcloud/cmek-ops-gcp.md` line 65 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N5HR2JX51QaNIswrvx2:-N5_aLF5ByDS8Wj4aG1T:b-ucfknh) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L65)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

I sent this feedback on Slack but don't want it to get lost -- the button isn't Add in this UI.
</blockquote></details>

Thanks!
___
*[`cockroachcloud/cmek-ops-gcp.md` line 70 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N5HRC0n8Q9i7iI68Y_k:-N5_aNb-DbuioURbZQXR:b-emxg48) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L70)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

This tripped me up even having read it! It would be nice if we could just trim it for them if they used the whole cluster ID. No action required, I suppose.
</blockquote></details>

It tripped me up as well!
___
*[`cockroachcloud/cmek-ops-gcp.md` line 84 at r4](https://reviewable.io/reviews/cockroachdb/docs/14287#-N5HRqC88A_CY_rs-1kw:-N5_aZo04m2SPxHCJ0tr:b-ilt3e3) ([raw file](https://github.com/cockroachdb/docs/blob/0494dc5ba164fd4028a311407c4f3339565521be/cockroachcloud/cmek-ops-gcp.md#L84)):*
<details><summary><i>Previously, mdlinville (Matt Linville) wrote…</i></summary><blockquote>

Is it worth mentioning here that multi-regional or global keys are fine? Below, is it worth stating that you could just as easily be importing as generating the key?
</blockquote></details>

I don't know. I feel like either SMEs should tell us (docs) what the suggested one or to ways are, or else we just leave it for them to figure out. I don't think it's our job to delimit the space of possibilities for our users (although I often feel the urge to do it), This feature seems to be currently in the latter phase, so leaving it for future refinement.

**Pull request merged**